### PR TITLE
Allow header to be anything in Tables.table

### LINKED ISSUE #199: adding Tables.table methods
@quinnj would you be OK to allow `AbstracVector{Symbol}` and `AbstractVector{<:AbstractString}` as a `header` in `Tables.table`.

relevant for a decision in https://github.com/JuliaData/DataFrames.jl/issues/2433#issuecomment-695050438.

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/Tables.jl/issues/199

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -240,6 +240,15 @@ end
     @test Tables.getcolumn(matrow, 1) == 1
     @test propertynames(mattbl) == propertynames(matrow) == [:Column1, :Column2, :Column3]
 
+    mattbl = Tables.table(mat, header=[:A, :B, :C])
+    @test Tables.columnnames(mattbl) == [:A, :B, :C]
+    mattbl = Tables.table(mat, header=view(["DUMMY", "A", "B", "C"], 2:4))
+    @test Tables.columnnames(mattbl) == [:A, :B, :C]
+    mattbl = Tables.table(mat, header=(["DUMMY", "A", "B", "C"] for i in 2:4))
```
Comment (bkamins) [inline test/runtests.jl]:
```suggestion
    mattbl = Tables.table(mat, header=(["DUMMY", "A", "B", "C"][i] for i in 2:4))
```

File: test/runtests.jl [hunk 2]
```diff
@@ -240,6 +240,15 @@ end
...
     @test propertynames(mattbl) == propertynames(matrow) == [:Column1, :Column2, :Column3]
 
+    mattbl = Tables.table(mat, header=[:A, :B, :C])
+    @test Tables.columnnames(mattbl) == [:A, :B, :C]
+    mattbl = Tables.table(mat, header=view(["DUMMY", "A", "B", "C"], 2:4))
+    @test Tables.columnnames(mattbl) == [:A, :B, :C]
+    mattbl = Tables.table(mat, header=(["DUMMY", "A", "B", "C"][i] for i in 2:4))
+    @test Tables.columnnames(mattbl) == [:A, :B, :C]
+    mattbl = Tables.table(mat, header=1:3)
+    @test Tables.columnnames(mattbl) == Symbol.(1:3)
```
Comment (bkamins) [inline test/runtests.jl:250]:
```suggestion
    @test Tables.columnnames(mattbl) == Symbol.(1:3)
    @test_throws ArgumentError Tables.table(mat, header=[:A, :B, :C, :D])
```

File: src/matrix.jl [hunk 3]
```diff
@@ -45,15 +45,19 @@ columnnames(m::MatrixTable) = names(m)
 """
     Tables.table(m::AbstractMatrix; [header::Vector{Symbol}])
 
-Wrap an `AbstractMatrix` (`Matrix`, `Adjoint`, etc.) in a `MatrixTable`, which satisfies
-the Tables.jl interface. This allows accesing the matrix via `Tables.rows` and
-`Tables.columns`. An optional keyword argument `header` can be passed as a `Vector{Symbol}`
-to be used as the column names. Note that no copy of the `AbstractMatrix` is made.
+Wrap an `AbstractMatrix` (`Matrix`, `Adjoint`, etc.) in a `MatrixTable`, which satisfies the
+Tables.jl interface. This allows accesing the matrix via `Tables.rows` and `Tables.columns`.
```
Comment (nalimilan) [inline src/matrix.jl]:
```suggestion
Tables.jl interface. This allows accessing the matrix via `Tables.rows` and `Tables.columns`.
```

File: src/matrix.jl [hunk 4]
```diff
@@ -45,15 +45,19 @@ columnnames(m::MatrixTable) = names(m)
 """
     Tables.table(m::AbstractMatrix; [header::Vector{Symbol}])
```
Comment (nalimilan) [inline src/matrix.jl]: Need to update the docstring here.

**Review by nalimilan — COMMENTED** [review]
Looks good.

File: src/matrix.jl (hunk 4 above)
Comment (bkamins) [reply src/matrix.jl]: done

**Review by quinnj — APPROVED** [review]

**Pull request merged**